python/dolfinx/wrappers/mesh.cpp

```diff
@@ -294,17 +284,30 @@ void mesh(py::module& m)
 
   m.def("locate_entities",
         [](const dolfinx::mesh::Mesh& mesh, int dim,
-           const std::function<std::vector<bool>(
-               const dolfinx::common::array2d<double>&)>& marker) {
-          return as_pyarray(dolfinx::mesh::locate_entities(mesh, dim, marker));
+           const std::function<py::array_t<bool>(
+               py::array_t<double, py::array::c_style>&)>& marker) {
+          auto cpp_marker = [&](const dolfinx::common::array2d<double>& x) {
```

Comment:
Let's be explicit with the capture - look like only `marker` needs to be captured here.